### PR TITLE
Add-linting-black-mypy

### PR DESCRIPTION
**Pull request by dinjazelena**

### CODE REVIEW

File: quinn/dataframe_helpers.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame, SparkSession
+from typing import Any
 
-from pyspark.sql import DataFrame, SparkSession
 from pyspark.sql.types import StructField, StructType
 
 
-def column_to_list(df: DataFrame, col_name: str) -> List[Any]:
+def column_to_list(df: DataFrame, col_name: str) -> list[Any]:
```
Comment (SemyonSinchenko) [inline quinn/dataframe_helpers.py:12]: Will it work with Python 3.7? Cause we are running tests fir spark 2 that requires python 3.7

File: quinn/dataframe_helpers.py [hunk 2]
```diff
@@ -18,8 +23,8 @@ def column_to_list(df: DataFrame, col_name: str) -> List[Any]:
 
 
 def two_columns_to_dictionary(
-    df: DataFrame, key_col_name: str, value_col_name: str
-) -> Dict[str, Any]:
+    df: DataFrame, key_col_name: str, value_col_name: str,
+) -> dict[str, Any]:
```
Comment (SemyonSinchenko) [inline quinn/dataframe_helpers.py:27]: The same concern about python 3.7

**Review by SemyonSinchenko — CHANGES_REQUESTED** [review]
I googled a little and found that we are able to write `list[..]` instead of `List[..]` starting from the version `3.9`. But we should support also `3.7` and `3.8` due to spark vetsions.

**Comment by SemyonSinchenko** [comment]
You can check the related part of CI [here](https://github.com/MrPowers/quinn/blob/main/.github/workflows/ci.yml#L41)

**Comment by dinjazelena** [comment]
Hey importing 'from __future__ import annotations' allows to use list[str] instead of List[str] in Python versions <3.9.
With this we can also use | instead of Union in Python versions <3.10.

Read more about it here:
https://docs.astral.sh/ruff/rules/future-rewritable-type-annotation/

**Comment by MrPowers** [comment]
Thank you for this.

**Comment by MrPowers** [comment]
@dinjazelena - the unit tests are failing for Spark 2.4.8: https://github.com/MrPowers/quinn/actions/runs/6444181927/job/17496859733. You know how to fix these?

**Comment by SemyonSinchenko** [comment]
We should put mkdocs in a separate dependency group

**Comment by dinjazelena** [comment]
@MrPowers , hey, yeah, this is dependency issue, so no worries. Will fix it today.

**Pull request merged**